src/test/net_tests.cpp

```diff
@@ -771,4 +773,129 @@ BOOST_AUTO_TEST_CASE(PoissonNextSend)
     g_mock_deterministic_tests = false;
 }
 
+std::vector<NodeEvictionCandidate> GetRandomNodeEvictionCandidates(const int n_candidates, FastRandomContext& random_context)
+{
+    std::vector<NodeEvictionCandidate> candidates;
+    for (int id = 0; id < n_candidates; ++id) {
+        candidates.push_back({/* id */ id,
+                              /* nTimeConnected */ static_cast<int64_t>(random_context.randrange(100)),
+                              /* nMinPingUsecTime */ static_cast<int64_t>(random_context.randrange(100)),
+                              /* nLastBlockTime */ static_cast<int64_t>(random_context.randrange(100)),
+                              /* nLastTXTime */ static_cast<int64_t>(random_context.randrange(100)),
+                              /* fRelevantServices */ random_context.randbool(),
+                              /* fRelayTxes */ random_context.randbool(),
+                              /* fBloomFilter */ random_context.randbool(),
+                              /* nKeyedNetGroup */ random_context.randrange(100),
+                              /* prefer_evict */ random_context.randbool(),
+                              /* m_is_local */ random_context.randbool()});
+    }
+    return candidates;
+}
+
+// Returns true if any of the node ids in node_ids are selected for eviction.
+bool IsEvicted(const std::vector<NodeEvictionCandidate>& candidates, const std::vector<NodeId> node_ids, FastRandomContext& random_context)
+{
+    std::vector<NodeEvictionCandidate> shuffled_candidates = candidates;
+    Shuffle(shuffled_candidates.begin(), shuffled_candidates.end(), random_context);
+    const Optional<NodeId> evicted_node_id = SelectNodeToEvict(shuffled_candidates);
```

Comment:
Any reason not to shuffle in place?

```suggestion
bool IsEvicted(std::vector<NodeEvictionCandidate>& candidates, const std::vector<NodeId> node_ids, FastRandomContext& random_context)
{
    Shuffle(candidates.begin(), candidates.end(), random_context);
    const Optional<NodeId> evicted_node_id = SelectNodeToEvict(candidates);
```